### PR TITLE
Add instance Hashable1 NonEmpty

### LINKED ISSUE #186: instance Hashable1 NonEmpty
I found this orphan instance in `hnix`. Could it be added here?

### PR DESCRIPTION
**Pull request by sjakobi**
Closes #186.

### CODE REVIEW

**Comment by sjakobi** [comment]
@phadej If this looks right, I'll make the corresponding PR for `semigroups`.

File: Data/Hashable/Class.hs [hunk 1]
```diff
@@ -826,6 +826,10 @@ instance Hashable1 Proxy where
 instance Hashable a => Hashable (NE.NonEmpty a) where
     hashWithSalt p (a NE.:| as) = p `hashWithSalt` a `hashWithSalt` as
 
+-- | @since UNRELEASED
+instance Hashable1 NE.NonEmpty where
+    liftHashWithSalt h salt (a NE.:| as) = liftHashWithSalt h (h salt a) as
+
 instance Hashable a => Hashable (Min a) where
```
Comment (phadej) [inline Data/Hashable/Class.hs:833]: I'd add instances to all `semigroups` types. `Min`, `Max` ...

And also to types from `Data.Monoid`: `All`, `Any`, `First`, `Last`, `Sum`, `Product`, `Alt`, `Ap`...

**Comment by sjakobi** [comment]
I have made a corresponding PR for hashable: https://github.com/ekmett/semigroups/pull/99

File: Data/Hashable/Class.hs (hunk 1 above)
Comment (sjakobi) [reply Data/Hashable/Class.hs:833]: I agree that these would be nice to have too.

I can't promise I'll work on this though.

**Comment by sjakobi** [comment]
> Then I request changes. We don't want to have a release per instance.

You don't have to make a release just for this one PR.

**Comment by phadej** [comment]
Having `master` is non-releasable state also forces maintainers to "clean up".

I'm sorry, but I have to ask for "a bit more" than you strictly need yourself. This way one can truly remove burden from maintainers shoulders.

**Comment by sjakobi** [comment]
> Having `master` is non-releasable state also forces maintainers to "clean up".

How would this PR put `master` in a non-releasable state?

> I'm sorry, but I have to ask for "a bit more" than you strictly need yourself. This way one can truly remove burden from maintainers shoulders.

As a maintainer and contributor in various projects I understand your concern. But your attempt to coerce me to share the maintenance burden unfortunately makes me less willing to contribute voluntarily.

Note that I haven't ruled out that I would contribute more instances. My main reason for not adding them right away was that I wanted to test the contribution process across the two repositories first.

**Comment by andreasabel** [comment]
@sjakobi wrote:
> Note that I haven't ruled out that I would contribute more instances. My main reason for not adding them right away was that I wanted to test the contribution process across the two repositories first.

I'd say this is a valid point.  As new contributor one is inclined to send a "test balloon" first to see whether PRs are welcome and handled in a responsible way.  Only if trust is established, one is willing to make a bigger investment.

For the maintainer it may be inconvenient to review and merge these "test balloon" PRs.  But business relationships usually start small until the mutual trust is established.  The situation with maintainer-contributor is similar.

In the end, we shouldn't forget that the maintainer-contributor relationship is asymmetric since the maintainer holds the keys and wields the power.  Having to ask for a merge is always a small humiliation.

It is a bit unfortunate that your communication didn't go as well as it could have gone.  Because I find @phadej's instantaneous attention to this PR exceptional.  You don't see such swift reactions a lot.

**Comment by phadej** [comment]
I'll fix my concerns in `master`.

I'll look around at other issues and will make a release ASAP. Thank you for your patience.

**Comment by sjakobi** [comment]
Thank you, @phadej! :+1:

**Pull request merged**